contracts/SpokePoolV3Periphery.sol

```diff
@@ -7,68 +7,137 @@ import { IERC20Permit } from "@openzeppelin/contracts/token/ERC20/extensions/IER
+ * pull its assets because the SpokePoolV3Periphery contract can be used to call
+ * any calldata on any exchange that the user wants to. By separating the contract that is approved to spend
+ * user funds from the contract that executes arbitrary calldata, the SpokePoolPeriphery does not
+ * need to validate the calldata that gets executed.
+ * @dev If this proxy didn't exist and users instead approved and interacted directly with the SpokePoolV3Periphery
+ * then users would run the unneccessary risk that another user could instruct the Periphery contract to steal
+ * any approved tokens that the user had left outstanding.
+ */
+contract SpokePoolPeripheryProxy is SpokePoolV3PeripheryProxyInterface, Lockable, MultiCaller {
+    using SafeERC20 for IERC20;
+    using Address for address;
+
+    // Flag set for one time initialization.
+    bool private initialized;
+
+    // The SpokePoolPeriphery should be deterministically deployed at the same address across all networks,
+    // so this contract should also be able to be deterministically deployed at the same address across all networks
+    // since the periphery address is the only initializer argument.
+    SpokePoolV3Periphery public spokePoolPeriphery;
+
+    error InvalidPeriphery();
+    error ContractInitialized();
+
+    /**
+     * @notice Construct a new Proxy contract.
+     * @dev Is empty and all of the state variables are initialized in the initialize function
+     * to allow for deployment at a deterministic address via create2, which requires that the bytecode
+     * across different networks is the same. Constructor parameters affect the bytecode so we can only
+     * add parameters here that are consistent across networks.
+     */
+    constructor() {}
+
+    /**
+     * @notice Initialize the SpokePoolPeripheryProxy contract.
+     * @param _spokePoolPeriphery Address of the SpokePoolPeriphery contract that this proxy will call.
+     */
+    function initialize(SpokePoolV3Periphery _spokePoolPeriphery) external nonReentrant {
+        if (initialized) revert ContractInitialized();
+        initialized = true;
+        if (!address(_spokePoolPeriphery).isContract()) revert InvalidPeriphery();
+        spokePoolPeriphery = _spokePoolPeriphery;
+    }
+
+    /**
+     * @notice Swaps tokens on this chain via specified router before submitting Across deposit atomically.
+     * Caller can specify their slippage tolerance for the swap and Across deposit params.
+     * @dev If swapToken or acrossInputToken are the native token for this chain then this function might fail.
+     * the assumption is that this function will handle only ERC20 tokens.
+     * @param swapAndDepositData Specifies the params we need to perform a swap on a generic exchange.
+     */
+    function swapAndBridge(SpokePoolV3PeripheryInterface.SwapAndDepositData calldata swapAndDepositData)
+        external
+        override
+        nonReentrant
```

Comment:
In `SpokePoolV3Periphery` this is `payable`. Should this one be as well?

Reply:
This is not payable since the proxy is only used for ERC20 approval based flows. If we swap and bridge with ETH, it is more gas-efficient to call the periphery directly, so we force that the user does that.